File: domain.availability/password/nodejs/domain_availability_pass.js
var https = require('https');
var querystring = require('querystring');

var url = "https://www.whoisxmlapi.com/"
    +"whoisserver/WhoisService?";

var parameters = {
    cmd: 'GET_DN_AVAILABILITY',
    domainName: 'google.com',
    username: 'YOUR_USERNAME',
    password: 'YOUR_PASSWORD',
    outputFormat: 'json'
};

url = url + querystring.stringify(parameters);

https.get(url, function (res) {
    const statusCode = res.statusCode;

    if (statusCode !== 200) {
        console.log('Request failed: '
            + statusCode
        );
    }

    var rawData = '';

    res.on('data', function(chunk) {
        rawData += chunk;
    });
    res.on('end', function () {
        try {
            var parsedData = JSON.parse(rawData);
            if (parsedData.DomainInfo) {
                console.log(
                    'Domain name: '
                    + parsedData.DomainInfo.domainName
                );
                console.log(
                    'Domain Availability: '
                    + parsedData.DomainInfo.domainAvailability
                );
            } else {
                console.log(parsedData);
            }
        } catch (e) {
            console.log(e.message);
        }
    })
}).on('error', function(e) {
    console.log("Error: " + e.message);
});

File: domain.availability/password/ruby/domain_availability_pass.rb
require 'open-uri'
require 'json'
require 'rexml/document'
require 'rexml/xpath'
require 'yaml'		# only needed to print the returned result in a very pretty way

########################
# Fill in your details #
########################
username = "YOUR_USERNAME"
password = "YOUR_PASSWORD"
domain = "google.com"

#######################
# Use a JSON resource #
#######################
format = "JSON"
url = 'https://www.whoisxmlapi.com/whoisserver/WhoisService?' + 'cmd=GET_DN_AVAILABILITY' + '&domainName=' + domain + '&username=' + username + '&password=' + password + '&outputFormat=' + format

# Open the resource
buffer = open(url).read

# Parse the JSON result
result = JSON.parse(buffer)

# Print out a nice informative string
puts "XML:\n" + result.to_yaml + "\n"

#######################
# Use an XML resource #
#######################
format = "XML"
url = 'http://www.whoisxmlapi.com/whoisserver/WhoisService?' + 'cmd=' + 'GET_DN_AVAILABILITY' + '&domainName=' + domain + '&username=' + username + '&password=' + password + '&outputFormat=' + format

# Open the resource
buffer = open(url).read

# Parse the XML result
result = REXML::Document.new(buffer)

# Get a few data members and make sure they aren't nil
if ((errorMessage = REXML::XPath.first(result, "/ErrorMessage/msg")) != nil)
	puts "JSON:\nErrorMessage:\n\t" + errorMessage.text
else
  domainAvailability = (domainAvailability = REXML::XPath.first(result, "/DomainInfo/domainAvailability")) == nil ? '' : domainAvailability.text
  domainName = (domainName = REXML::XPath.first(result, "/DomainInfo/domainName")) == nil ? '' : domainName.text
	puts "JSON:\n---\n" + "  domainAvailability: " + domainAvailability + "\n  domainName: " + domainName
end
